bindings/java/src/main/native/jucx_common_def.cc

```diff
@@ -204,11 +223,25 @@ UCS_PROFILE_FUNC(jobject, process_request, (request, callback), void *request, j
 
     // If request is a pointer set context callback and rkey.
     if (UCS_PTR_IS_PTR(request)) {
-        if (callback != NULL) {
-            ((struct jucx_context *)request)->callback = env->NewGlobalRef(callback);
+        struct jucx_context *ctx = (struct jucx_context *)request;
+        ucs_spin_lock(&ctx->lock);
+        if (ucs_unlikely(ctx->status != UCS_INPROGRESS && callback != NULL)) {
```

Comment:
ok, now this logic can be simplified and remove some code duplication as follows:
```
if (ctx->status == UCS_INPROGRESS) {
  // request not completed yes by another thread, install user callback
  if (callback != NULL) {
    ctx->callback = env->NewGlobalRef(callback);
  }
 ctx->jucx_request = env->NewGlobalRef(jucx_request);
} else {
  // request already completed by another thread
  if (callback != NULL) {
    jucx_call_callback(callback, jucx_request, ctx->status);
  }
  jucx_context_reset(ctx);
  ucp_request_free(request);
}
```
```